tracing-appender/src/rolling.rs

```diff
@@ -558,17 +562,100 @@ impl Inner {
                     .unwrap_or(0),
             ),
             rotation,
+            max_files,
         };
         Ok((inner, writer))
     }
 
+    fn prune_old_logs(&self, max_files: usize) {
+        let format;
+        match self.rotation {
+            Rotation::MINUTELY => {
+                format = format_description::parse("[year]-[month]-[day]-[hour]-[minute]")
+                    .expect("Unable to create a formatter; this is a bug in tracing-appender");
+            }
+            Rotation::HOURLY => {
+                format = format_description::parse("[year]-[month]-[day]-[hour]")
+                    .expect("Unable to create a formatter; this is a bug in tracing-appender");
+            }
+            Rotation::DAILY => {
+                format = format_description::parse("[year]-[month]-[day]")
+                    .expect("Unable to create a formatter; this is a bug in tracing-appender");
+            }
+            Rotation::NEVER => {
+                unreachable!("maximum log files feature is not available for `Rotation::NEVER`!")
+            }
+        }
```

Comment:
it's kind of unfortunate that we parse a new format description every time `prune_old_logs` is called, and we also parse them separately in the `join_date` function, here: https://github.com/tokio-rs/tracing/blob/08c44200ae7d966439f673bffc69f549b3dd94d1/tracing-appender/src/rolling.rs#L500-L506

this is potentially inefficient, as it will allocate to store the format description, and also parses the string each time, although this isn't a big deal as these functions aren't called that frequently. additionally, though, it means that the format description strings are defined in two places now, both here in `prune_old_logs` and separately in `join_date`...if we change one of the format descriptions, we have to change the other one as well, or `prune_old_logs` won't work anymore.

what if we did a small refactor to move the format description to a field on `Inner`, and create it only once when constructing a new appender? then, we could just use `self.date_format` in both this method and in `join_date`.

Reply:
When I first opened the PR 2 months ago, my initial attempt was to add `format` to `Inner`. However it was unsuccessful due to the ***lifetime*** necessity. 

I'll call it a day for now (it's been a long day for me), and I can give this problem another go at the weekend.

Reply:
if the format descriptions are always `&'static str`s (which they are), the lifetime of the `FormatItem`s should also be `'static`. if you like, i can open a PR against this branch to make that change?

Reply:
sure, I don't exactly recall at which step I got stuck 2 months ago. Maybe this time it will be a success. But if you have the time, I won't reject this offer :)

Reply:
okay, i've gone ahead and opened https://github.com/ozgunozerk/tracing-keep-last-n-logs/pull/1 with that change --- let me know what you think!

Reply:
wonderful!